.github/workflows/ci.yml

```diff
@@ -0,0 +1,443 @@
+        run: |
+          $rubyArchitecture = (ruby -e "puts RbConfig::CONFIG['arch']").Trim()
+          $gemVersion = (Get-Content VERSION).Trim()
+          $gemToInstall = "./tiny_tds-$gemVersion-$rubyArchitecture.gem"
+          
+          Write-Host "Looking to install $gemToInstall"
+          gem install --local "$gemToInstall"
+          
+          ruby -e "require 'tiny_tds'; puts TinyTds::Gem.root_path"
+          exit $LASTEXITCODE
+
+  test-windows-mingw:
+    needs:
+      - cross-compile
+    strategy:
+      fail-fast: false
+      matrix:
+        mssql-version:
+          #- 2017
+          - 2019
+          - 2022
+        ruby-version:
+          - 2.7
+          - '3.0'
+
+    name: Test on Windows (MingW)
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+
+      - name: Download precompiled gem
+        uses: actions/download-artifact@v4
+        with:
+          name: gem-x64-mingw32
+
+      - name: Install native gem and restore cross-compiled code from it
+        shell: pwsh
+        run: |
+          $rubyArchitecture = (ruby -e "puts RbConfig::CONFIG['arch']").Trim()
+          $gemVersion = (Get-Content VERSION).Trim()
+          $gemToInstall = "./tiny_tds-$gemVersion-$rubyArchitecture.gem"
+
+          Write-Host "Looking to install $gemToInstall"
+          gem install --local --install-dir=./tmp "$gemToInstall"
+
+          # Restore precompiled code
+          $source = (Resolve-Path ".\tmp\gems\tiny_tds-$gemVersion-$rubyArchitecture\lib\tiny_tds").Path
+          $destination = (Resolve-Path ".\lib\tiny_tds").Path
+          Get-ChildItem $source -Recurse -Exclude "*.rb" | Copy-Item -Destination {Join-Path $destination $_.FullName.Substring($source.length)}
+
+          # Restore ports
+          Copy-Item -Path ".\tmp\gems\tiny_tds-$gemVersion-$rubyArchitecture\ports" -Destination "." -Recurse
+
+      - name: Setup MSSQL
+        uses: potatoqualitee/mssqlsuite@v1.7
+        with:
+          install: sqlengine, sqlclient
+          version: ${{ matrix.mssql-version }}
+          sa-password: c0MplicatedP@ssword
+          show-log: true
+
+      - name: Setup MSSQL database
+        shell: pwsh
+        run: |
+          & sqlcmd -S localhost -U sa -P "c0MplicatedP@ssword" -i ./test/sql/db-create.sql
+          & sqlcmd -S localhost -U sa -P "c0MplicatedP@ssword" -i ./test/sql/db-login.sql
+
+      - name: Install toxiproxy-server
+        shell: pwsh
+        run: |
+          choco install toxiproxy-server --version=2.5.0 -y
+          Start-Process toxiproxy-server
+
+      - name: Test gem
+        shell: pwsh
+        run: bundle exec rake test
+        env:
+          TOXIPROXY_HOST: "localhost"
+
+      - name: Test Summary
+        uses: test-summary/action@v2
+        with:
+          paths: "test/reports/TEST-*.xml"
+        if: always()
```

Comment:
I'm curious, how do test failures show up here?

Reply:
I made a branch with a test that fails intentionally. here is the result: https://github.com/andyundso/tiny_tds/actions/runs/8900358242

here is also a screenshot:

![image](https://github.com/rails-sqlserver/tiny_tds/assets/7010698/e51b92c9-abdf-4d74-a188-ab3347ba40f8)